File: Examples/output/cpp/googletest/example2.cpp
// Copyright (c) 2019 ...

// Other bespoke headers
#include "../cppscenarios/example2.h"

// Third party headers
#include "gtest/gtest.h"

namespace Cornichon::Accumulator2
{
  class TestFixture : public ::testing::Test
  {
  protected:
    void SetUp() override
    {
    }

    void TearDown() override
    {
    }
  };

  TEST_F(TestFixture, AddOneOther)
  {
    Scenarios::AddOneOther scenario;
    scenario.GivenAnInitial6();
    scenario.WhenYouAddA5();
    scenario.ThenTheResultIs11();
  }

  TEST_F(TestFixture, AddTwoOthers)
  {
    Scenarios::AddTwoOthers scenario;
    scenario.GivenAnInitial6();
    scenario.WhenYouAddA8();
    scenario.WhenYouAddA4();
    scenario.ThenTheResultIs18();
  }
}

File: Examples/output/py/pytests/example2.py
import pytest
from example2_scenarios import *


def test_add_one_other():
    """Gherkin DSL test"""
    scenario = Scenarios.AddOneOther()
    scenario.GivenAnInitial6()
    scenario.WhenYouAddA5()
    scenario.ThenTheResultIs11()


def test_add_two_others():
    """Gherkin DSL test"""
    scenario = Scenarios.AddTwoOthers()
    scenario.GivenAnInitial6()
    scenario.WhenYouAddA8()
    scenario.WhenYouAddA4()
    scenario.ThenTheResultIs18()
